### ISSUE TITLE
Manually set horizon weights to different windows

### ISSUE THREAD

**Issue by Newaij0** [Nixtla/neuralforecast#939]
### Description

To my understanding, currently the implementation of horizon weights is by adding a fixed horizon_weight list when initializing the loss function. 

```python
horizon_weight = np.hstack((np.ones(10,)*2, np.ones(10,), np.ones(10,)*2))
horizon = 30 

model1 = TFT(h = horizon, 
             input_size = horizon*3,
             hidden_size = 64, 
             n_head = 4,
             loss = MAE(horizon_weight = horizon_weight),  ##
             valid_loss = MAE(), # MAE()
             learning_rate = 0.001,
             dropout = 0.1,
             attn_dropout = 0.1,
             futr_exog_list = futr_exog,
             hist_exog_list = hist_exog,
             stat_exog_list = stat_exog,
             max_steps = 600,
             val_check_steps = 50,  
             scaler_type = 'robust',
             batch_size = 128)
```
However in the case I'm dealing with, the loss during holidays is considered more important, not the distance between predicted timestamp and current timestamp. 

I add a column to the input DataFrame representing the weight of each timestamp and try to modify the base window class to extract weights for different time windows. Is there any way to adding weight more easily? 

Thanks

### Use case

Setting different horizon weight to each window by simply adding a weight column in the input DataFrame.

**Comment by elephaint** [comment]
Thanks for the suggestion. While we consider it, options to account for different sample importances are:

- Use a different loss function, e.g. MSE if holidays typically show higher target values. MSE will penalize errors on those higher values to a larger extent than MAE.
- This is a bit iffy, but you can scale the target by the sample weights yourself and scale back with the correct weights for prediction when predicting a holiday or non-holiday. In that case, the forecasting task would be to forecast the importance-weighted time series rather than the actual time series.
